storage/tests/test_isolation.rs

```diff
@@ -122,3 +130,335 @@ fn g0_update_conflicts_fail() {
+    let value_10 = ByteArray::inline([10], 1); // value_<k><t> = Key k written by Transaction t
+    let value_12 = ByteArray::inline([12], 1);
+    let value_20 = ByteArray::inline([20], 1);
+    let value_22 = ByteArray::inline([22], 1);
+    let storage_path = create_tmp_dir();
+    let storage = setup_storage(&storage_path);
+
+    let snapshot_setup = storage.open_snapshot_write();
+    snapshot_setup.put_val(key_1.to_owned(), ByteArray::copy(value_10.bytes()));
+    snapshot_setup.put_val(key_2.to_owned(), ByteArray::copy(value_20.bytes()));
+    let result_snapshot = snapshot_setup.commit();
+    assert!(result_snapshot.is_ok());
+
+    let snapshot_1 = storage.open_snapshot_write();
+    let snapshot_2 = storage.open_snapshot_write();
+
+    let read_11 = snapshot_1.get::<128>(key_1);
+    snapshot_2.put_val(key_1.to_owned(), ByteArray::copy(value_12.bytes()));
+    snapshot_2.put_val(key_2.to_owned(), ByteArray::copy(value_22.bytes()));
+    let read_21 = snapshot_1.get::<128>(key_2);
+
+    assert_eq!(value_10.bytes(), read_11.unwrap().bytes());
+    assert_eq!(value_20.bytes(), read_21.unwrap().bytes());
+
+    let result_1 = snapshot_1.commit();
+    let result_2 = snapshot_2.commit();
+}
+
+#[test]
+fn g2_write_skew_disjoint_read() {
+    init_logging();
+    let key_1: &StorageKey<'_, 48> = &StorageKey::Reference(StorageKeyReference::new(KEYSPACE_ID, ByteReference::new(&KEY_1)));
+    let key_2: &StorageKey<'_, 48> = &StorageKey::Reference(StorageKeyReference::new(KEYSPACE_ID, ByteReference::new(&KEY_2)));
+    let value_0 = ByteArray::inline([10], 1); // value_<k><t> = Key k written by Transaction t
+    let value_1 = ByteArray::inline([11], 1);
+    let storage_path = create_tmp_dir();
+    let storage = setup_storage(&storage_path);
+
+    let snapshot_setup = storage.open_snapshot_write();
+    snapshot_setup.put_val(key_1.to_owned(), ByteArray::copy(value_0.bytes()));
+    snapshot_setup.put_val(key_2.to_owned(), ByteArray::copy(value_0.bytes()));
+    let result_snapshot = snapshot_setup.commit();
+    assert!(result_snapshot.is_ok());
+
+    let snapshot_1 = storage.open_snapshot_write();
+    let snapshot_2 = storage.open_snapshot_write();
+
+    let read_21 = snapshot_1.get::<128>(key_2);
+    if (read_21.unwrap().bytes()[0] == 0) {
+        snapshot_1.put_val(key_1.to_owned(), ByteArray::copy(value_1.bytes()));
+    }
+    let read_12 = snapshot_2.get::<128>(key_1);
+    if (read_12.unwrap().bytes()[0] == 0) {
+        snapshot_2.put_val(key_2.to_owned(), ByteArray::copy(value_1.bytes()));
+    }
+
+    let result_1 = snapshot_1.commit();
+    let result_2 = snapshot_2.commit();
+
+    assert!(result_1.is_ok() && result_2.is_ok());
+    let reader_after = storage.open_snapshot_read();
+    let sum: u8 = reader_after.get::<128>(key_1).unwrap().bytes()[0] + reader_after.get::<128>(key_2).unwrap().bytes()[0];
+    assert!(sum <= 1);
```

Comment:
t1 (t2) reads key_1 (key_2), and writes key_2 (key_1) in a way that key_1 xor key_2 = 1
This fails, as expected